### DISCUSSION TITLE
issue: `shouldUnregister: true`: watched value with defaultValues return `undefined` when using conditional rendering or reset. (different from #10096)

### DISCUSSION THREAD

**Discussion by Hengry** [react-hook-form/react-hook-form#10180] [Q&A] [answered]
### Version Number

7.43.7

### Codesandbox/Expo snack

https://codesandbox.io/s/compassionate-bell-e0ccls?file=/src/App.tsx

### Steps to reproduce

1. Click on 'open'
2. See ```controlled undefined controlledValue``` in the console, while the expected one should be ```controlled controlledValue controlledValue```
3. Click on 'reset'
4. See ```controlled undefined newValue``` in the first time re-render, while the expected one should be ```controlled newValue newValue```

all of the above would be as same as the expected one with `shouldUnregister: false`

### Expected behaviour

The result of step2 should be ```controlled controlledValue controlledValue```,  
and the result of step4 should be ```controlled newValue newValue```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
If `shouldUnregister` is not working as expected, use `unregsiter` method in your own use-cases.

**Comment by Hengry** [comment]
@bluebill1049  But I think the one who is out of expectation is `watch`. If the watched field is a `Controller` in `shouldUnregister: true`, then the result of first time re-render after `reset` would falsely be `undefined` instead of the value in reset, also the result of applying `defaultValue` is `undefined` instead of the defaultValue.

In both of the cases, the return of `getValues` would be correct, so I think it might be some issues in `watch`.

**Comment by bluebill1049** [comment, marked as answer]
There is a lot going on with check input mounts or unmounts, I would probably recommend using unregister if your use case. `watch` API has the condition to check mount or unmount when you invoke reset if will remove all input references and wait until all inputs get re-render and remounted.

**Reply by dheerajsinghnagdali** [reply]
When will this issue be fixed?
